src/components/DropdownFilterContainer.ts

```diff
@@ -0,0 +1,147 @@
+import { Component, ReactElement, createElement } from "react";
+import { findDOMNode } from "react-dom";
+import * as dijitRegistry from "dijit/registry";
+import * as classNames from "classnames";
+import * as dojoLang from "dojo/_base/lang";
+import * as dojoConnect from "dojo/_base/connect";
+
+import { DropdownFilter, DropdownFilterProps } from "./DropdownFilter";
+import { ValidateConfigs } from "./ValidateConfigs";
+
+export interface DropdownFilterContainerProps {
+    entity: string;
+    mxform: mxui.lib.form._FormBase;
+    targetGridName: string;
+    values: ValueProps[];
+}
+
+export interface ValueProps {
+    caption: string;
+    comparison: comparisonOptions;
+    attribute: string;
+    value: string;
+    constraint: string;
+    filterMethod: filterMethodOptions;
+}
+
+type filterMethodOptions = "equals" | "contains";
+export type comparisonOptions = "static" | "XPath";
+type HybridConstraint = Array<{ attribute: string; operator: string; value: string; path?: string; }>;
+
+export interface ListView extends mxui.widget._WidgetBase {
+    datasource: {
+        xpathConstraints: string;
+    };
+    _datasource: {
+        _constraints: HybridConstraint | string;
+        _entity: string;
+        _setSize: number;
+        atEnd: () => boolean;
+        _pageSize: number;
+    };
+    _loadMore: () => void;
+    _onLoad: () => void;
+    _renderData: () => void;
+    update: () => void;
+}
+
+interface DropdownFilterState {
+    findingWidget: boolean;
+    targetGrid?: ListView;
+    targetNode?: HTMLElement;
+    validationPassed?: boolean;
+}
+
+export default class DropdownFilterContainer extends Component<DropdownFilterContainerProps, DropdownFilterState> {
+
+    constructor(props: DropdownFilterContainerProps) {
+        super(props);
+
+        this.state = { findingWidget: true };
+        this.handleChange = this.handleChange.bind(this);
+        dojoConnect.connect(props.mxform, "onNavigation", this, dojoLang.hitch(this, this.initDropdownFilter));
+    }
+
+    render() {
+        return createElement("div",
```

Comment:
no need to create an additional div, only do that when there is a validation error.

Reply:
Done